tools/run.py

```diff
@@ -24,11 +25,18 @@ def register(command):
         default=False,
         help="Run the specified program using valgrind",
     )
-    command.add_argument("args", nargs="+", help="the command and any arguments that should be used for the execution")
+
+    # Find all role files
+    fnames = glob(os.path.join("roles", "**", "*.role"), recursive=True)
+    # Strip roles/ and .role from file paths
+    roles = [f[6:-5] for f in fnames]
```

Comment:
This removes any subdirectories between roles/ and the .role files from the file paths so roles in fake/ and test/ can't be run

Reply:
Hmm thats a good point, in that case you should use `os.path.splitext(f.replace(os.path.join(b.project_dir, "roles", ""), ""))[0]`

It's better to use 

1. an absolute path (`b.project_dir`) because you don't _have_ to run the b script from the project root directory
2. `os.path` functions to try and ensure os-agnosticism